include/odp/api/spec/ipsec.h

```diff
@@ -772,6 +772,76 @@ typedef struct odp_ipsec_sa_param_t {
 
 } odp_ipsec_sa_param_t;
 
+/** IPv4 address size */
+#define ODP_IP4ADDR_SIZE 4
+
+/** IPv6 address size */
+#define ODP_IP6ADDR_SIZE 16
+
+/**
+ * Retrieve IPSEC SA information
+ */
+typedef struct odp_ipsec_sa_info_t {
+	/** Copy of IPSEC Security Association (SA) parameters
+	 *
+	 * All pointers will be set to NULL.
+	 */
+	odp_ipsec_sa_param_t param;
+
+	/** Recent value of SA lifetime limits */
+	odp_ipsec_lifetime_t lifetime_recent;
```

Comment:
I do not understand this. Lifetime limits do not change. They are set at SA creation time and it is not possible to alter them later. So the "recent" limits are the same as the initial ones in SA parameters and this field is redundant.

But maybe you think that this tells the *remaining* lifetime, i.e. the original packet and byte lifetimes subtracted by the number of packets and bytes processed, respectively? That would seem to duplicate the functionality of the statistics API that returns the number of packets processed (not bytes in current proposal, IIRC, but it can be added if needed). So even with this interpretation I do not see why this field would be needed here.

Reply:
Remaining lifetime you just described was the idea. But agreed that the same can be achieved with the statistics API. I will drop the lifetime remaining field from the SA info structure.